index.js

```diff
@@ -89,24 +92,54 @@ router
     "This endpoint is used to determine which metrics (collections) are available to the data source."
   );
 
+const seriesQuery = function(collection, start, end, interval) {
+  const { filterExpression, dateField, valueField, dateExpression,
+          valueExpression } = cfg;
+
+  let filterSnippet = aql.literal(filterExpression
+    ? `FILTER ${filterExpression}`
+    : "");
+
+  let dateSnippet = aql.literal(dateExpression
+    ? `LET d = ${dateExpression}`
+    : `LET d = doc["${dateField}"]`);
+
+  let valueSnippet = aql.literal(valueExpression
+    ? `LET v = ${valueExpression}`
+    : `LET v = doc["${valueField}"]`);
```

Comment:
FWIW This allows entering arbitrary AQL, including data modification queries.

That said, I'd suggest doing it this way:

```js
let valueSnippet = aql`LET v = ${
  valueExpression
  ? aql.literal(valueExpression)
  : aql`doc[${valueField}]`
}`;
```

Also note that ```
aql``
``` is equivalent to `aql.literal("")`.

Reply:
Using the `aql` template tag for the value field avoids problems with weird field names.

Reply:
done